core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregatorUtils.java

```diff
@@ -48,8 +50,13 @@ public static DeduplicateAggregationDefinition generateAggregationDefinition(Rel
     String inputRelationName = ctx.getInputRelationNames().stream().findFirst().orElse(null);
     Schema inputSchema = inputRelationName != null ? ctx.getInputSchema(inputRelationName) : null;
 
-    Optional<ExpressionFactory<String>> expressionFactory = ctx.getEngine().
-      getExpressionFactory(StringExpressionFactoryType.SQL);
+    Optional<ExpressionFactory<String>> expressionFactory = getExpressionFactory(ctx, inputSchema);
+
+    // If the expression factory is not present, this aggregation cannot be handled by the plugin.
+    if (!expressionFactory.isPresent()) {
+      return null;
```

Comment:
Is caller ready to handle nulls?  Please mark method as `Nullable`. 
nit: It would be great to have a javadoc as this is public method

Reply:
I've updated the method and added the javadoc.